lib/primer/yard/lookbook_pages_backend.rb

```diff
@@ -0,0 +1,194 @@
+# frozen_string_literal: true
+
+# :nocov:
+module Primer
+  module Yard
+    # A single Lookbook page.
+    class LookbookPage
+      include DocsHelper
+
+      PREVIEW_MAP = {
+        Primer::Alpha::TextField => [:single_text_field_form, :multi_text_field_form],
+        Primer::Alpha::TextArea => [],
+        Primer::Alpha::Select => [:select_list_form],
+        Primer::Alpha::MultiInput => [:multi_input_form],
+        Primer::Alpha::RadioButton => [:radio_button_with_nested_form],
+        Primer::Alpha::RadioButtonGroup => [:radio_button_group_form],
+        Primer::Alpha::CheckBox => [:check_box_with_nested_form],
+        Primer::Alpha::CheckBoxGroup => [:check_box_group_form],
+        Primer::Alpha::SubmitButton => [:submit_button_form],
+        Primer::Alpha::FormButton => [:submit_button_form]
+      }.freeze
+
+      attr_reader :component_ref, :backend, :docs
+
+      def initialize(component_ref, backend, docs)
+        @component_ref = component_ref
+        @backend = backend
+        @docs = docs
+      end
+
+      def page_id
+        @page_id ||= docs.short_name.dasherize.underscore.tap do |page_id|
+          page_id << "_input" unless page_id.end_with?("_input")
+        end
+      end
+
+      def generate
+        path = File.expand_path(
+          File.join(
+            *%w[.. .. .. demo test components docs forms inputs],
```

Comment:
Should we unify these with the `previews/` folder? Just thinking about dotcom's lookbook grabbing that folder for displaying previews. Combining might allow us to serve these docs in dotcom as well

Reply:
Oh yeah that's a great idea! When you say "unify," do you mean writing them to another directory at the same level as previews/ or _inside_ previews/?

Reply:
Inside I think but separate from the actual previews. Like previews/pages/ ?